src/ReplaceImage.js

```diff
@@ -0,0 +1,35 @@
+function ReplaceImage(ref,selector,steps) {
+  if(!ref.options.inBrowser) return; // This isn't for Node.js
+  this_ = ref;
+  var input = document.querySelectorAll(selector);
+  var images = [];
+  for (i = 0; i < input.length; i++)
+    if (input[i] instanceof HTMLImageElement) images.push(input[i]);
+  for (i in images) {
+    the_image = images[i];
+    var url = images[i].src;
+    var ext = url.split('.').pop();
+
+    var xmlHTTP = new XMLHttpRequest();
+    xmlHTTP.open('GET', url, true);
+    xmlHTTP.responseType = 'arraybuffer';
+    xmlHTTP.onload = function(e) {
+      var arr = new Uint8Array(this.response);
+      var raw = String.fromCharCode.apply(null,arr);
+      var base64 = btoa(raw);
+      var dataURL="data:image/"+ext+";base64," + base64;
+      make(dataURL);
+    };
+
+    if(url.substr(0,11).toLowerCase()!="data:image/") xmlHTTP.send();
+    else make(url);
+
+    function make(url) {
```

Comment:
PROS:
* May bypass some CORS restrictions.
* Will also work on browsers not supporting Canvas. (read: IE < 9)

CONS:
* An extra HTTP request each time.

I feel this needs more testing (to find out whether the premise of bypassing CORS restrictions is actually valid). What do you say?